crates/roc_host/src/lib.rs

```diff
@@ -614,57 +634,73 @@ pub extern "C" fn roc_fx_fileReadBytes(roc_path: &RocList<u8>) -> RocResult<RocL
 }
 
 #[no_mangle]
-pub extern "C" fn roc_fx_fileReader(roc_path: &RocList<u8>) -> RocResult<u64, RocStr> {
+pub extern "C" fn roc_fx_fileReader(
+    roc_path: &RocList<u8>,
+    size: u64,
+) -> RocResult<RocBox<()>, RocStr> {
     match File::open(path_from_roc_path(roc_path)) {
-        Ok(file) => READERS.with(|reader_thread_local| {
-            let mut readers = reader_thread_local.borrow_mut();
-
-            // monotonically increasing index so each reader has a unique identifier
-            let new_index = READER_INDEX.fetch_add(1, Ordering::SeqCst);
-
-            let buf_reader = BufReader::new(file);
-
-            readers.insert(new_index, Rc::new(RefCell::new(Box::new(buf_reader))));
+        Ok(file) => {
+            let buf_reader = if size > 0 {
+                BufReader::with_capacity(size as usize, file)
```

Comment:
Not a blocker for this PR, but I do wonder whether we want to separate buffered and unbuffered streams.

Offering an unbuffered option means I can write pure Roc code that results in the bare minimum number of syscalls for I/O with no allocations involved, which in turn means I can approach C levels of performance as close as possible if I really want to optimize things!

Reply:
Long term, I think we should dump bufreader and just do it in roc.